Pilot/pilotCommands.py

```diff
@@ -654,6 +697,14 @@ def __setFlavour( self ):
       if 'BOINC_HOST_NAME' in os.environ:
         self.boincHostName = os.environ['BOINC_HOST_NAME']
 
+    # Pilot reference is given explicitly in environment
+    if 'PILOT_UUID' in os.environ:
+      pilotRef = os.environ['PILOT_UUID']
```

Comment:
That will set pilotRef to None if the environment variable isn't present. Ok for PILOT_UUID but not for some of the batch/grid systems which the code is trying to identify one by one in an order that matters (IIUC). I think this code should be moved to pilotTools where both pilot commands and pilot logging can use it. It can be tidied up at that point, but it still has to determine the flavour there too, using the logic currently in this function.

Reply:
Right, so better `pilotRef = os.environ.get('PILOT_UUID', pilotRef)`

Reply:
Can you please either do the move yourself, or add the above comment directly in the code?